### PR TITLE
Fix sorting in featuregates

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
Make sure featuregates are sorted

### CODE REVIEW

File: pkg/operator/target_config_reconciler_v311_00.go [hunk 1]
```diff
@@ -145,11 +146,16 @@ func managePod_v311_00_to_latest(client coreclientv1.ConfigMapsGetter, recorder
...
 	// check for feature gates from feature lister.
 	featureGates := checkForFeatureGates(featureGateLister)
+	for featureGateName := range featureGates {
+		sortedFeatureGates = append(sortedFeatureGates, featureGateName)
+	}
+	sort.Strings(sortedFeatureGates)
 	allFeatureGates := ""
-	for featureGate, status := range featureGates {
-		allFeatureGates = allFeatureGates + "," + fmt.Sprintf("%v=%v", featureGate, status)
+	for _, featureGateName := range sortedFeatureGates {
```
Comment (deads2k) [inline pkg/operator/target_config_reconciler_v311_00.go]: for future reference, `range sets.StringKeySet(featureGates).List()`

**Comment by sjenning** [comment]
https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_cluster-kube-scheduler-operator/99/pull-ci-openshift-cluster-kube-scheduler-operator-master-e2e-aws/403/artifacts/e2e-aws/clusteroperators.json
this is not the (whole) fix
```
        {
            "apiVersion": "config.openshift.io/v1",
            "kind": "ClusterOperator",
            "metadata": {
                "creationTimestamp": "2019-04-12T19:05:27Z",
                "generation": 1,
                "name": "kube-scheduler",
                "resourceVersion": "29923",
                "selfLink": "/apis/config.openshift.io/v1/clusteroperators/kube-scheduler",
                "uid": "eedfaa5a-5d55-11e9-80c1-12d89a5eb096"
            },
            "spec": {},
            "status": {
                "conditions": [
                    {
                        "lastTransitionTime": "2019-04-12T19:43:33Z",
                        "reason": "NodeInstallerFailingInstallerPodFailed",
                        "status": "True",
                        "type": "Failing"
                    },
                    {
                        "lastTransitionTime": "2019-04-12T19:05:44Z",
                        "message": "Progressing: 1 nodes are at revision 140; 1 nodes are at revision 128; 1 nodes are at revision 123",
                        "reason": "Progressing",
                        "status": "True",
                        "type": "Progressing"
                    },
                    {
                        "lastTransitionTime": "2019-04-12T19:11:13Z",
                        "message": "Available: 3 nodes are active; 1 nodes are at revision 140; 1 nodes are at revision 128; 1 nodes are at revision 123",
                        "reason": "AsExpected",
                        "status": "True",
                        "type": "Available"
                    },
```

**Comment by ravisantoshgudimetla** [comment]
@sjenning and @deads2k - Not sure, if the previous run used older image(image before my change). In any case, I want to run this test.

/test e2e-aws

**Comment by ravisantoshgudimetla** [comment]
Flakes not related to control plane.
/test e2e-aws

**Comment by deads2k** [comment]
result looks much better

/lgtm

**Comment by ravisantoshgudimetla** [comment]
/retest

flake again with market-place operators..

> Not sure, if the previous run used older image(image before my change). 

@deads2k, any reason why it failed for the first time?

**Pull request merged**